### PR TITLE
Return correct result counts for select:file.owners

### PR DESCRIPTION
**Pull request by eseliger**
In the previous fix, I forgot that select:file.owners could also drop results entirely and not just potentially expand them to multiple results. So here we go, another fix for the same problem.

## Test plan

Verified result counts are now correct.

### CODE REVIEW

File: internal/search/job/jobutil/job.go [hunk 1]
```diff
@@ -192,12 +192,13 @@ func NewBasicJob(inputs *search.Inputs, b query.Basic) (job.Job, error) {
...
+		if isSelectOwnersSearch(b, inputs.Features) {
 			// the select owners job is ran separately as it requires state and can return multiple owners from one match.
-			if inputs.Features.CodeOwnershipSearch && sp.Root() == filter.File && len(sp) == 2 && sp[1] == "owners" {
-				basicJob = codeownershipjob.NewSelectOwnersSearch(basicJob)
+			basicJob = codeownershipjob.NewSelectOwnersSearch(basicJob)
+		} else {
+			if v, _ := b.ToParseTree().StringValue(query.FieldSelect); v != "" {
+				sp, _ := filter.SelectPathFromString(v) // Invariant: select already validated
+				basicJob = NewSelectJob(sp, basicJob)
 			}
```
Comment (eseliger) [inline internal/search/job/jobutil/job.go:202]: A question I also asked on your other PR @leonore: Do we need to chain both jobs here? The results look the same to me 🤔 if yes, I'll revert the else branch here.

Comment (leonore) [reply internal/search/job/jobutil/job.go:202]: I think we have this check in the select owners job:
```go
		mm, ok := m.(*result.FileMatch)
		if !ok {
			continue
		}
```
which means it should work fine and just ignore any other types of matches, so we probably can do either one or the other.

File: internal/search/job/jobutil/job.go [hunk 2]
```diff
@@ -543,6 +548,18 @@ func isOwnershipSearch(b query.Basic, features *search.Features) (include, exclu
...
+	// Check if we have a select field.
+	if v, _ := b.ToParseTree().StringValue(query.FieldSelect); v != "" {
+		// Parse the select field path.
+		sp, _ := filter.SelectPathFromString(v) // Invariant: select already validated
+		// If the feature flag is enabled, and the filter is for file.owners, this is
+		// a select:file.owners search and we should apply special limits.
+		return features.CodeOwnershipSearch && sp.Root() == filter.File && len(sp) == 2 && sp[1] == "owners"
+	}
+	return false
+}
```
Comment (leonore) [inline internal/search/job/jobutil/job.go:561]:
```suggestion
func isSelectOwnersSearch(sp filter.SelectPath, features *search.Features) bool {
		// If the feature flag is enabled, and the filter is for file.owners, this is
		// a select:file.owners search and we should apply special limits.
		return features.CodeOwnershipSearch && sp.Root() == filter.File && len(sp) == 2 && sp[1] == "owners"
	}
	return false
}
```

File: internal/search/job/jobutil/job.go (hunk 1 above)
Comment (leonore) [reply internal/search/job/jobutil/job.go:202]: github isn't letting me do suggestions nicely, but what about this (with my other suggestion as well):
```go
       	sp, _ := filter.SelectPathFromString(v) // Invariant: select already validated
		if isSelectOwnersSearch(sp, inputs.Features) {
			// the select owners job is ran separately as it requires state and can return multiple owners from one match.
			basicJob = codeownershipjob.NewSelectOwnersSearch(basicJob)
		} else {
             basicJob = NewSelectJob(sp, basicJob)
        }
```

Comment (eseliger) [reply internal/search/job/jobutil/job.go]: alright, now's that? :)

**Review by leonore — APPROVED** [review]

**Pull request merged**